applicationinsights/django/common.py

```diff
@@ -0,0 +1,93 @@
+import collections
+
+from django.conf import settings
+import applicationinsights
+
+ApplicationInsightsSettings = collections.namedtuple("ApplicationInsightsSettings", [
+    "ikey",
+    "channel_settings",
+    "use_operation_url",
+    "record_view_arguments"])
+
+ApplicationInsightsChannelSettings = collections.namedtuple("ApplicationInsightsChannelSettings", [
+    "send_interval",
+    "send_time",
+    "endpoint"])
+
+def load_settings():
+    if hasattr(settings, "APPLICATION_INSIGHTS"):
+        config = settings.APPLICATION_INSIGHTS
+    elif hasattr(settings, "APPLICATIONINSIGHTS"):
+        config = settings.APPLICATIONINSIGHTS
+    else:
+        config = {}
+
+    if not isinstance(config, dict):
+        config = {}
+
+    if settings.DEBUG:
+        ikey = config.get("debug_ikey")
+        endpoint = config.get("debug_endpoint", config.get("endpoint"))
+    else:
+        ikey = config.get("ikey")
+        endpoint = config.get("endpoint")
+
+    return ApplicationInsightsSettings(
+        ikey=ikey,
+        use_operation_url=config.get("use_operation_url", False),
+        record_view_arguments=config.get("record_view_arguments", True),
+        channel_settings=ApplicationInsightsChannelSettings(
+            endpoint=endpoint,
+            send_interval=config.get("send_interval"),
+            send_time=config.get("send_time")))
+
+saved_clients = {}
+saved_channels = {}
+
+def create_client(aisettings=None):
+    global saved_clients, saved_channels
+
+    if aisettings is None:
+        aisettings = load_settings()
+
+    if aisettings in saved_clients:
+        return saved_clients[aisettings]
+
+    channel_settings = aisettings.channel_settings
+
+    if channel_settings in saved_channels:
+        channel = saved_channels[channel_settings]
+    else:
+        if channel_settings.endpoint is not None:
+            sender = applicationinsights.channel.AsynchronousSender(service_endpoint_uri=channel_settings.endpoint)
```

Comment:
Right, what I'm saying is if it is none the line will still work as it will just run
`applicationinsights.channel.AsynchronousSender(service_endpoint_uri=None)`